beacon_node/lighthouse_network/src/peer_manager/peerdb.rs

```diff
@@ -1166,6 +1160,47 @@ impl<E: EthSpec> PeerDB<E> {
     }
 }
 
+fn compute_custody_subnets_from_enr<E: EthSpec>(
+    enr: &Enr,
+    peer_id: &PeerId,
+    spec: &ChainSpec,
+    log: &Logger,
+) -> HashSet<DataColumnSubnetId> {
+    let node_id = enr.node_id().raw().into();
+
+    // fallback to `custody_requirement` if `csc` is invalid
+    let custody_subnet_count = enr.custody_subnet_count::<E>().unwrap_or_else(|e| {
+        // This handles the scenario where `csc` is not found or not decodable.
+        // TODO(das): Should we penalise peer?
+        debug!(
+            log,
+            "No valid csc found in peer ENR";
+            "info" => "Falling back to default custody requirement",
+            "peer_id" => %peer_id,
+            "enr" => enr.to_base64(),
+            "error" => ?e
+        );
```

Comment:
Both of this debug statements should also increase a metric

Reply:
Is this really necessary?

Reply:
so we get a count of how many peers have invalid / missing csc in their ENR? Does this metrics help us in anyway other than trying to catch a bug? 

I feel like a log is suffice if we just want to catch a bug.

Reply:
I've deleted the computation from ENR code in peerdb after a chat with @AgeManning - it's best to avoid doing error handling / peer reporting logic in `PeerDB::update_connection_state`, as the function focuses on handling various connection state transitions, and is not the right place for this.

I've updated the subnet predicate to not perform any fallback and just return `false` in the error case, so in theory we shouldn't be connecting to these peers with bad `csc` values via discovery. 

We can just rely on metadataV3 for `csc` as we always get them from peers and this should simplify things.